src/shared/components/Post/FeaturedVideo.js

```diff
@@ -0,0 +1,109 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { inject } from 'mobx-react';
+import styled, { css } from 'styled-components';
+import { Helmet } from 'react-helmet';
+import SharedCount from './SharedCount';
+import ReadingTime from './ReadingTime';
+
+const FeaturedVideo = ({
+  type,
+  id,
+  src,
+  sharedCountPosition,
+  readingTimePosition,
+  isAmp,
+}) => {
+  const match =
+    src.match(/\/embed\/([\d\w-]+)/) || src.match(/\/([\d\w-]+?)\?/);
+  const videoId = match ? match[1] : null;
+  return (
+    <Container>
+      {isAmp ? (
+        <Fragment>
+          <Helmet>
+            <script
+              async=""
+              custom-element="amp-youtube"
+              src="https://cdn.ampproject.org/v0/amp-youtube-0.1.js"
+            />
+          </Helmet>
+          <AmpContainer>
+            <amp-youtube layout="fill" data-videoid={videoId} />
+          </AmpContainer>
+        </Fragment>
+      ) : (
+        <Iframe title="featured-video" src={src} allowFullScreen="" />
+      )}
+      {(sharedCountPosition === 'featured-image' ||
```

Comment:
All this code is being duplicated, maybe makes more sense to create a `<Featured />` component that inside will load either `<FeaturedImage />` or `<FeaturedYoutube />` or even just `Image` or `Youtube` if any of this components exists already.

Reply:
@orballo I've made some changes in addition to what you proposed. Check it out.